### DISCUSSION TITLE
Widmungsdarstellung des geoland-Service Vorarlberg wird mit "Grauschleier" dargestellt

### DISCUSSION THREAD

**Discussion by faulgurke** [e-netze/webgis-community#150] [Q&A]
Mit freundlicher Unterstützung von Christian habe ich den geoland-Service von gView4 auf gView5 umgestellt. Dabei ist uns aufgefallen, dass die Farben der Widmungskategorien einen Grauschleier aufweisen (also sehr unbunt sind).
(Das ist übrigens der Service aus dem Diskussionsforum #147)

Darstellung Service:
![geoland_widmung_service](https://github.com/gis-eni/webgis-forum/assets/100088983/da0d0fda-44ea-4853-b0a0-0493f9914443)

Die Darstellung in der Konfigurationsdatei (.mxl) ist dagegen tadellos.

Darstellung Konfiguration:
![geoland_widmung_konfig](https://github.com/gis-eni/webgis-forum/assets/100088983/6b74e4c6-0848-443e-bebb-8653372525e5)

Wenn man in der Konfiguration die Transparenz der Darstellung entfernt, werden die Farben wieder "bunt" dargestellt (allerdings ist dann die gewünschte Transparenz perdü).
Vielleicht hat das mit der Umstellung von GdiPlus auf SkiaSharp zu tun?
Lösungsvorschläge?

**Comment by gis-eni** [comment]
Ist eigentlich keine Thema für das WebGIS Forum.
JPG oder PNG?
Was ist, wenn du auf Gdi+ umstellst?

**Comment by gis-eni** [comment]
Folgendes hab ich noch heraus gefunden.
Man kann in der ``mapserver.config`` einstellen, mit welcher Engine gerendert wird:

Kann es sein, dass das bei dir so ausschaut?

``` json
"graphics": {
  "rendering": "skia",
  "encoding": "gdiplus"
}
```

Das ist die default Einstellung. Das Rendering ist in der Regel mit Skia besser und funkt auch unter Linux. Das Encoding (also das schreiben/speichern der Bilder) wurde per default einmal ``gdiplus`` gesetzt, weil das ein wenig performanter ist und kleiner Bilder erzeugt. Allerdings, scheint hier dann die Transparenz anders interpretiert zu werden :(
Stell einmal beides auf ``skia`` (dann AppPool durchstarten). Dann sollte das Ergebnis passen.

**Reply by faulgurke** [reply]
Meinst du die Datei mapserver.json?
Da war bei mir bezüglich "graphics" gar kein Eintrag vorhanden.
Aber jetzt sieht es gut aus.

**Reply by gis-eni** [reply]
ja die mein ich... ```json``, ``config``, same, same ;)

Wenn nix drin steht, zieht er anscheinend die default Einstellungen und das ist diese Mischform von oben.
Zur Info:
in zukünftigen Version wird es dann auch wieder mit den Standard Einstellungen "gut" aussehen. Anscheinend haben die beiden Libraries ein anderes Verständnis von "transparenz". Es gibt aber abhilfe.
Grundsätzlich spricht natürlich auch nix dagegen alles mit "skia" fahren, wenn die Performance passt. Und wenn ihr in Zukunft sowieso eher Linux/Kubernetes verwendet wollt, wird GDI+ vielleicht eh nicht gehen
